app/services/follow_remote_account_service.rb

```diff
@@ -29,20 +29,24 @@ def call(uri)
     return Account.find_local(confirmed_username) if TagManager.instance.local_domain?(confirmed_domain)
 
     confirmed_account = Account.find_remote(confirmed_username, confirmed_domain)
-    return confirmed_account unless confirmed_account.nil?
-
-    Rails.logger.debug "Creating new remote account for #{uri}"
+    if confirmed_account.nil?
+      Rails.logger.debug "Creating new remote account for #{uri}"
+
+      domain_block = DomainBlock.find_by(domain: domain)
+      account = Account.new(username: confirmed_username, domain: confirmed_domain)
+      account.suspended   = true if domain_block && domain_block.suspend?
+      account.silenced    = true if domain_block && domain_block.silence?
+      account.private_key = nil
+    else
+      account = confirmed_account
+    end
 
-    domain_block = DomainBlock.find_by(domain: domain)
+    account.last_webfingered_at = Time.now.utc
```

Comment:
I believe AR and postgres handle this, but if other parts of our code are doing it then go ahead.